core/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -231,7 +250,7 @@ public static int getNumberOfDelayedUnassigned(long unassignedShardsAllocatedTim
     }
 
     /**
-     * Finds the smallest delay expiration setting of an unassigned shard. Returns 0 if there are none.
+     * Finds the smallest delay expiration setting in milliseconds of an unassigned shard. Returns 0 if there are none.
      */
     public static long findSmallestDelayedAllocationSetting(Settings settings, ClusterState state) {
         long nextDelaySetting = Long.MAX_VALUE;
```

Comment:
can we change this method semantics to only consider shards that are actively delayed - i.e. getLastComputedLeftDelayNanos > 0 ?